modules/pallets/collator-balances/src/lib.rs

```diff
@@ -0,0 +1,272 @@
+	#[pallet::storage]
+	pub type Bonded<T: Config> =
+		StorageMap<_, Blake2_128Concat, T::AccountId, T::Balance, ValueQuery>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The account does not have enough unreserved funds to bond the requested amount.
+		InsufficientBalance,
+	}
+
+	/// Makes use of the underlying implementation provided by the `NativeCurrency` i.e
+	/// pallet-balances
+	impl<T: Config> Currency<T::AccountId> for Pallet<T> {
+		type Balance = T::Balance;
+		type PositiveImbalance = PositiveImbalanceOf<T>;
+		type NegativeImbalance = NegativeImbalanceOf<T>;
+
+		fn total_balance(who: &T::AccountId) -> Self::Balance {
+			T::NativeCurrency::total_balance(who)
+		}
+
+		fn can_slash(who: &T::AccountId, value: Self::Balance) -> bool {
+			T::NativeCurrency::can_slash(who, value)
+		}
+
+		fn total_issuance() -> Self::Balance {
+			T::NativeCurrency::total_issuance()
+		}
+
+		fn minimum_balance() -> Self::Balance {
+			T::NativeCurrency::minimum_balance()
+		}
+
+		fn burn(amount: Self::Balance) -> Self::PositiveImbalance {
+			T::NativeCurrency::burn(amount)
+		}
+
+		fn issue(amount: Self::Balance) -> Self::NegativeImbalance {
+			T::NativeCurrency::issue(amount)
+		}
+
+		fn free_balance(who: &T::AccountId) -> Self::Balance {
+			T::NativeCurrency::free_balance(who)
+		}
+
+		fn slash(
+			who: &T::AccountId,
+			value: Self::Balance,
+		) -> (Self::NegativeImbalance, Self::Balance) {
+			T::NativeCurrency::slash(who, value)
+		}
+
+		fn transfer(
+			source: &T::AccountId,
+			dest: &T::AccountId,
+			value: Self::Balance,
+			existence_requirement: frame_support::traits::ExistenceRequirement,
+		) -> DispatchResult {
+			T::NativeCurrency::transfer(source, dest, value, existence_requirement)
+		}
+
+		fn ensure_can_withdraw(
+			who: &T::AccountId,
+			amount: Self::Balance,
+			reasons: WithdrawReasons,
+			new_balance: Self::Balance,
+		) -> DispatchResult {
+			T::NativeCurrency::ensure_can_withdraw(who, amount, reasons, new_balance)
+		}
+
+		fn deposit_into_existing(
+			who: &T::AccountId,
+			value: Self::Balance,
+		) -> Result<Self::PositiveImbalance, DispatchError> {
+			T::NativeCurrency::deposit_into_existing(who, value)
+		}
+
+		fn deposit_creating(who: &T::AccountId, value: Self::Balance) -> Self::PositiveImbalance {
+			T::NativeCurrency::deposit_creating(who, value)
+		}
+
+		fn withdraw(
+			who: &T::AccountId,
+			value: Self::Balance,
+			reasons: WithdrawReasons,
+			liveness: ExistenceRequirement,
+		) -> Result<Self::NegativeImbalance, DispatchError> {
+			T::NativeCurrency::withdraw(who, value, reasons, liveness)
+		}
+
+		fn make_free_balance_be(
+			who: &T::AccountId,
+			balance: Self::Balance,
+		) -> SignedImbalance<Self::Balance, Self::PositiveImbalance> {
+			T::NativeCurrency::make_free_balance_be(who, balance)
+		}
+	}
+
+	/// The custom implementation of the `ReservableCurrency` trait.
+	impl<T: Config> ReservableCurrency<T::AccountId> for Pallet<T> {
```

Comment:
Don't think we need this to be a pallet, we should look into how `pallet-conviction-voting` handles locks on balances and just implement that for ReservableCurrency